x/ecocredit/server/utils/test_utils.go

```diff
@@ -1,18 +1,124 @@
 package utils
 
 import (
-	"github.com/golang/mock/gomock"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	"github.com/regen-network/regen-ledger/x/ecocredit/mocks"
+	"github.com/tendermint/tendermint/abci/types"
+	"google.golang.org/protobuf/proto"
 )
 
-// ExpectParamGet is a helper function that sets up an expected mock call for the provided type.
-func ExpectParamGet[T any](obj *T, paramKeeper *mocks.MockParamKeeper, key []byte, times int) {
-	gmAny := gomock.Any()
-	var expectedType T
-	paramKeeper.EXPECT().Get(gmAny, key, &expectedType).Do(func(_ sdk.Context, _ []byte, param *T) {
-		*param = *obj
-	}).Times(times)
+// MatchEvent matches the values in a proto message struct to the attributes in a sdk.Event.
+func MatchEvent(event any, emitted sdk.Event) error {
```

Comment:
This is looking good. Can we move to `types/testutil` so that we can reuse in the data module?

Reply:
moved [refactor: move event matching code to types/testutil](https://github.com/regen-network/regen-ledger/pull/1394/commits/130922dd8b7d1745ef2101cf4b65e1e69db08e5b)